plugins/modules/web_infrastructure/bugzilla.py

```diff
@@ -0,0 +1,227 @@
+    type: str
+    returned: always
+'''
+
+import base64
+import json
+import sys
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.urls import fetch_url
+
+
+# Handle request by ansible lib
+def handle_request(url, timeout, data=None, method='GET'):
+    data = json.dumps(data)
+    response, info = fetch_url(module, url, data=data, method=method, timeout=timeout,
+                               headers={'Content-Type': 'application/json'})
+    if info['status'] not in (200, 201, 204):
+        module.fail_json(msg=info['msg'])
+    body = response.read()
+    ret = {}
+    if body:
+        ret = json.loads(body)
+    return ret
+
+
+def comment(restbase, user, passwd, params):
+    data = {
+        'comment': params['comment'],
+        'Bugzilla_login': user,
+        'Bugzilla_password': passwd,
+    }
+    url = restbase + '/bug/' + params['bug'] + '/comment'
+    ret = handle_request(url, params['timeout'], data, 'POST')
+    return ret
+
+
+def fetch(restbase, user, passwd, params):
+    url = restbase + '/bug/' + params['bug'] + "?Bugzilla_login=" + user + "&Bugzilla_password=" + passwd
+    ret = handle_request(url, params['timeout'])
+    return ret
+
+
+def transition(restbase, user, passwd, params):
+    data = {
+        'status': params['status'],
+        'Bugzilla_login': user,
+        'Bugzilla_password': passwd,
+    }
+    url = restbase + '/bug/' + params['bug']
+    ret = handle_request(url, params['timeout'], data, 'PUT')
+    return ret
+
+
+# Some parameters are required depending on the operation:
+OP_REQUIRED = dict(comment=['bug', 'comment'],
+                   fetch=['bug'],
+                   transition=['bug', 'status'])
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            uri=dict(required=True),
+            operation=dict(choices=['comment', 'fetch', 'transition'],
+                           aliases=['command'], required=True),
+            username=dict(required=True),
+            password=dict(required=True, no_log=True),
+            comment=dict(),
+            status=dict(),
+            bug=dict(),
+            timeout=dict(type='float', default=10),
+        ),
+        supports_check_mode=False
+    )
+
+    op = module.params['operation']
+
+    # Check we have the necessary per-operation parameters
+    missing = []
+    for parm in OP_REQUIRED[op]:
+        if not module.params[parm]:
+            missing.append(parm)
+    if missing:
+        module.fail_json(msg="Operation %s require the following missing parameters: %s" % (op, ",".join(missing)))
+
+    uri = module.params['uri']
+    user = module.params['username']
+    passwd = module.params['password']
+
+    if not uri.endswith('/'):
+        uri = uri + '/'
+    restbase = uri + 'rest'
+
+    try:
+        # Lookup the corresponding method for this operation.
+        thismod = sys.modules[__name__]
+        method = getattr(thismod, op)
```

Comment:
Please don't write code like this. Use a map with function entries instead if you really want that.

Reply:
Ack, changed in the local codes, will push later.